spec/services/openid_connect_redirector_spec.rb

```diff
@@ -35,6 +35,16 @@
   end
 
   describe '#validate' do
+    context 'with a redirect_uri that spoofs a hostname' do
+      let(:redirect_uri) { 'https://example.com.evilish.com/' }
+
+      it 'is invalid' do
+        redirector.validate
+        expect(errors[:redirect_uri]).
+          to include(t('openid_connect.authorization.errors.redirect_uri_no_match'))
+      end
+    end
+
```

Comment:
This doesn't test that the controller actually validates the redirect_uri from this class. Could would write an end-to-end feature spec please?